src/mnode/mnode-rpc.cpp

```diff
@@ -1216,159 +1217,184 @@ UniValue governance(const UniValue& params, bool fHelp)
-				"passphrase for new key cannot be empty!");
+R"(pastelid newkey "passphrase"
+passphrase for new key cannot be empty!)");
 
         UniValue resultObj(UniValue::VOBJ);
-
-        std::string pastelID = CPastelID::CreateNewLocalKey(strKeyPass);
-        resultObj.pushKV("pastelid", move(pastelID));
+        auto keyMap = CPastelID::CreateNewPastelKeys(std::move(strKeyPass));
+        if (keyMap.empty())
+            throw runtime_error("Failed to generate new PastelID and associated keys");
+        resultObj.pushKV("pastelid", std::move(keyMap.begin()->first));
+        resultObj.pushKV(RPC_KEY_LEGROAST, std::move(keyMap.begin()->second));
         return resultObj;
     }
     if (PASTELID.IsCmd(RPC_CMD_PASTELID::importkey))
     {
         if (params.size() < 2 || params.size() > 3)
             throw JSONRPCError(RPC_INVALID_PARAMETER,
-				"pastelid importkey \"key\" <\"passphrase\">\n"
-				"Import PKCS8 encrypted private key (EdDSA448) in PEM format. Return PastelID base58-encoded if \"passphrase\" provided."
-			);
+R"(pastelid importkey "key" <"passphrase">
+Import PKCS8 encrypted private key (EdDSA448) in PEM format. Return PastelID base58-encoded if "passphrase" provided.)");
     
         throw runtime_error("\"pastelid importkey\" NOT IMPLEMENTED!!!");
     
         //import
         //...
 
-        //validate and geenrate pastelid
+        //validate and generate pastelid
         if (params.size() == 3)
         {
             SecureString strKeyPass;
             strKeyPass.reserve(100);
             strKeyPass = params[2].get_str().c_str();
 
-            if (strKeyPass.length() < 1)
+            if (strKeyPass.empty())
                 throw runtime_error(
-					"pastelid importkey <\"passphrase\">\n"
-					"passphrase for imported key cannot be empty!");
+R"(pastelid importkey <"passphrase">
+passphrase for imported key cannot be empty!)");
         }
 
         UniValue resultObj(UniValue::VOBJ);
         return resultObj;
     }
+
+    // list all locally stored PastelIDs and associated public keys
     if (PASTELID.IsCmd(RPC_CMD_PASTELID::list))
     {
         UniValue resultArray(UniValue::VARR);
 
-        const auto pastelIDs = CPastelID::GetStoredPastelIDs();
-        for (const auto & p: pastelIDs)
+        auto mapIDs = CPastelID::GetStoredPastelIDs(false);
+        for (auto& [sPastelID, sLegRoastPubKey] : mapIDs)
         {
             UniValue obj(UniValue::VOBJ);
-            obj.pushKV("PastelID", p);
-            resultArray.push_back(move(obj));
+            obj.pushKV("PastelID", std::move(sPastelID));
+            obj.pushKV(RPC_KEY_LEGROAST, std::move(sLegRoastPubKey));
+            resultArray.push_back(std::move(obj));
         }
 
         return resultArray;
     }
+
+    // sign text with the internally stored private key associated with the PastelID (ed448 or legroast).
     if (PASTELID.IsCmd(RPC_CMD_PASTELID::sign))
     {
-        if (params.size() != 4)
+        if (params.size() < 4)
```

Comment:
if (params.size() < 4 || params.size()>5)

Reply:
well, let's say we have new UI version communicating with old pasteld daemon. In this case it theoretically can send more than 5 parameters. This old version will simply won't work in this case.

Reply:
yes, but think about next UI version and even newer API.
We actually need here only first 5 parameters and don't care what goes next.
For backward compatibility with all future versions - it's good to use < > param count comparisons.

Reply:
agree